libs/core/src/lib/wizard/wizard.component.ts

```diff
@@ -0,0 +1,211 @@
+        });
+        if (stepsArray.length > 1) {
+            let currentStepIndex = 0,
+                stepToHide;
+            if (currentStep) {
+                currentStepIndex = stepsArray.indexOf(currentStep);
+            }
+            currentStepIndex > (stepsArray.length - 1) / 2
+                ? (stepToHide = stepsArray[0])
+                : (stepToHide = stepsArray[stepsArray.length - 1]);
+            stepToHide.elRef.nativeElement.classList.add(STEP_NO_LABEL_CLASS);
+            stepToHide.elRef.nativeElement.classList.add(STEP_STACKED_CLASS);
+            this._setStackedTop(currentStep);
+        }
+    }
+
+    /** @hidden */
+    private _setStackedTop(currentStep: WizardStepComponent): void {
+        this.steps.forEach((step, index) => {
+            step.elRef.nativeElement.classList.remove(STEP_STACKED_TOP_CLASS);
+            if (this.steps.toArray()[index + 1] === currentStep) {
+                step.elRef.nativeElement.classList.add(STEP_STACKED_TOP_CLASS);
+            }
+        });
+    }
+
+    /** @hidden */
+    private _resetStepClasses(): void {
+        this.steps.forEach((step) => {
+            step.elRef.nativeElement.classList.remove(STEP_STACKED_TOP_CLASS);
+            step.elRef.nativeElement.classList.remove(STEP_STACKED_CLASS);
+            step.elRef.nativeElement.classList.remove(STEP_NO_LABEL_CLASS);
+        });
+    }
+
+    /** @hidden */
+    private _handleStepOrStatusChanges(): void {
+        this._setContentTemplate();
+        this._shrinkWhileAnyStepIsTooNarrow();
+        this._cdRef.detectChanges();
+    }
+
+    /** @hidden */
+    private _shrinkWhileAnyStepIsTooNarrow(): void {
+        this._resetStepClasses();
+        let i = 0;
+        while (this._anyStepIsTooNarrow() && i < this.steps.length - 1) {
+            i++;
+            this._wizardShrinking();
+        }
+    }
+
+    /** @hidden */
+    private _anyStepIsTooNarrow(): boolean {
+        let foundNarrowStep = false;
+        this.steps.forEach((step) => {
+            if (step.elRef.nativeElement.clientWidth < STEP_MIN_WIDTH) {
+                foundNarrowStep = true;
+            }
+        });
+        return foundNarrowStep;
+    }
+
+    /** @hidden */
+    private _stepClicked(clickedStep: WizardStepComponent): void {
+        this.steps.forEach((step) => {
+            if (step === clickedStep) {
+                step.status = 'current';
+                step.statusChange.emit('current');
+            } else if (step !== clickedStep) {
+                if (this.steps.toArray().indexOf(step) < this.steps.toArray().indexOf(clickedStep)) {
+                    step.status = 'completed';
+                    step.statusChange.emit('completed');
+                } else if (this.steps.toArray().indexOf(step) > this.steps.toArray().indexOf(clickedStep)) {
+                    step.status = 'upcoming';
+                    step.statusChange.emit('upcoming');
```

Comment:
would be better to use constants or string enum for statuses (current/completed/upcoming)

Reply:
If we wanted to use an enum here then the application developer would have to import the StepType enum rather than just passing strings as inputs and I'd prefer to leave that option as it's what we're doing with every other component (see various fdType examples).  I will use constants